clusterloader2/pkg/measurement/util/controlled_pods_indexer.go

```diff
@@ -65,12 +161,75 @@ func NewControlledPodsIndexer(podsInformer coreinformers.PodInformer, rsInformer
 		return nil, fmt.Errorf("failed to register indexer: %w", err)
 	}
 
-	return &ControlledPodsIndexer{
-		podsIndexer: podsInformer.Informer().GetIndexer(),
-		podsSynced:  podsInformer.Informer().HasSynced,
-		rsIndexer:   rsInformer.Informer().GetIndexer(),
-		rsSynced:    rsInformer.Informer().HasSynced,
-	}, nil
+	cpi := &ControlledPodsIndexer{
+		podsIndexer:           podsInformer.Informer().GetIndexer(),
+		podsSynced:            podsInformer.Informer().HasSynced,
+		rsIndexer:             rsInformer.Informer().GetIndexer(),
+		rsSynced:              rsInformer.Informer().HasSynced,
+		rsUidToState:          RsUidToState{},
+		deploymentUidToRsUids: DeploymentUidToRsUids{},
+	}
+
+	podsInformer.Informer().AddEventHandler(
+		cache.ResourceEventHandlerFuncs{
+			AddFunc: func(obj interface{}) {
+				cpi.uidsLock.Lock()
+				defer cpi.uidsLock.Unlock()
+
+				cpi.handleIncomingPodAddEvent(obj)
+			},
+			UpdateFunc: func(oldObj, newObj interface{}) {
+				cpi.uidsLock.Lock()
+				defer cpi.uidsLock.Unlock()
+
+				cpi.handleIncomingPodAddEvent(newObj)
+				cpi.handleIncomingPodDeleteEvent(oldObj)
+
+			},
+			DeleteFunc: func(obj interface{}) {
+				cpi.uidsLock.Lock()
+				defer cpi.uidsLock.Unlock()
+
+				cpi.handleIncomingPodDeleteEvent(obj)
+			},
+		},
+	)
+	rsInformer.Informer().AddEventHandler(
+		cache.ResourceEventHandlerFuncs{
+			AddFunc: func(obj interface{}) {
+				cpi.uidsLock.Lock()
+				defer cpi.uidsLock.Unlock()
+
+				metaAccessor, err := meta.Accessor(obj)
+				if err != nil {
+					return
+				}
+
+				rsUID := metaAccessor.GetUID()
```

Comment:
Can this be a part of handleIncomingRSAddEvent?

Reply:
Yup, done.